File: ejercicios hechos/Python/ejercicios repeticiones/act3.py
#Programa: act3.py
#Propósito: Algoritmo que pida caracteres e imprima ‘VOCAL’ si son vocales y ‘NO VOCAL’ en caso contrario, el programa termina cuando se introduce un espacio.
#Autor: Jose Manuel Serrano Palomo.
#Fecha: 29/10/2019
#
#Variables a usar:
# 
# 
#
#Algoritmo:
# Repetiremos hasta que solo sea un carácter
# Mientras no haya espacio
#   Si caracter no es espacio
#   Si es A,E,I,O,U o a,e,i,o,u --> Mostrar "Es vocal"
#   Sino mostrar "No es vocal"
#   Repetiremos hasta que solo sea un carácter

print("Leer carácteres hasta que solo haya un espacio")
print("-----------------------------------------------\n")

# Leemos los datos
caracter = str(input("Introduce un carácter: "))

while len(caracter) != 1:
    caracter = str(input("Introduce un carácter: "))

# Realizamos la comprobacion
while caracter != " ":
    if caracter == "A" or caracter == "E" or caracter == "I" or caracter == "O" or caracter == "U" or caracter == "a" or caracter == "e" or caracter == "i" or caracter == "o" or caracter == "u":
        print("Es una vocal")
    else:
        print("No es una vocal")
        
    caracter = str(input("Introduce un carácter: "))

File: ejercicios terminados/Java/estructuras repetitivas/act3.java
/*Programa: act3.java
 * 
 *Propósito: Algoritmo que pida caracteres e imprima ‘VOCAL’ si son vocales
 *  y ‘NO VOCAL’ en caso contrario, el programa termina cuando se introduce 
 * un espacio.
 * 
 *Autor: Jose Manuel Serrano Palomo.
 * 
 *Fecha: 10/11/2019
 *
 *Análisis:
 * En este programa debemos comprobar que caracter es vocal o no
 * y lo mostramos por pantalla
 *
 *Diseño
 * 1. Repetir -> Leer carácter hasta que sea un sólo carácter
 * 2. Mientras no sea espacio
  * 3. Si car no es el espacio
  * 4. Si es A,E,I,O,U o a,e,i,o,u -> Mostrar "Es vocal"
  * 5.  Si no mostrar "No es vocal"
  * 6. Repetir -> Leer carácter hasta que sea un sólo carácter
*/
import java.util.Scanner;
public class act3 {
    public static void main(String[] args) {
        System.out.println("Programa para determinar si un caracter es vocal");
        System.out.println("------------------------------------------------");
        Scanner s = new Scanner(System.in);
        String car;

        do {
          System.out.print("Introduce un carácter: ");
          car = s.nextLine();
        } while (car.length()!=1);

        while (!car.equals(" ")) {
          if (car.toUpperCase().equals("A") || car.toUpperCase().equals("E") || car.toUpperCase().equals("I") || car.toUpperCase().equals("O") || car.toUpperCase().equals("U")) {
            System.out.println("VOCAL");
          } else {
            System.out.println("NO VOCAL");
          }
          do {
            System.out.print("Introduce un carácter: ");
            car = s.nextLine();
          } while (car.length()!=1);
        }
    }
}
